src/gui/painting/qprintengine_pdf.cpp

```diff
@@ -833,7 +833,31 @@ class jpg_header_reader {
 
 };
 
+bool QPdfEnginePrivate::smartCompressDeflate(const char *source, int size, QByteArray& compressed, unsigned long maxSize)
+{
+    if (maxSize == 0) {
+        maxSize = ::compressBound(size);
+    }
 
+    const unsigned long initialBufferSize = 10 * 1024 * 1024;
+
+    unsigned long destLen = min(initialBufferSize, maxSize);
+
+    while (true) {
+        compressed.resize(destLen);
+        if (Z_OK == ::compress(reinterpret_cast<Bytef *>(compressed.data()), &destLen, reinterpret_cast<const Bytef *>(source), size)) {
+            compressed.truncate(destLen);
+            break;
+        }
+        if (static_cast<unsigned long>(compressed.size()) >= maxSize) {
+            compressed.clear();
+            return false;
+        }
+        destLen = min(static_cast<unsigned long>(compressed.size()) * 2, maxSize);
+    }
+
+    return true;
```

Comment:
shouldn't this check `maxSize` with `compressed.size()` e.g. as in line 930 `(uLongf)destLen < target`?